beacon-chain/state/state-native/state_trie.go

```diff
@@ -530,6 +533,7 @@ func (b *BeaconState) Copy() state.BeaconState {
 		nextSyncCommittee:                   b.nextSyncCommitteeVal(),
 		latestExecutionPayloadHeader:        b.latestExecutionPayloadHeaderVal(),
 		latestExecutionPayloadHeaderCapella: b.latestExecutionPayloadHeaderCapellaVal(),
+		historicalSummaries:                 b.historicalSummariesVal(),
```

Comment:
if we are sharing fields for historical summaries with other state instances, then this is incorrect. This performs a whole copy of the field, when we should just be sharing the reference like how it is done for historical roots:
```
historicalRoots: b.historicalRoots,
```